compiler/rustc_infer/src/infer/sub.rs

```diff
@@ -209,6 +209,18 @@ impl<'tcx> TypeRelation<'tcx> for Sub<'_, '_, 'tcx> {
         self.fields.infcx.super_combine_consts(self, a, b)
     }
 
+    fn effects(
+        &mut self,
+        a: ty::Effect<'tcx>,
+        b: ty::Effect<'tcx>,
+    ) -> RelateResult<'tcx, ty::Effect<'tcx>> {
+        match (a.val, b.val) {
+            // Any effect matches the rigid "on" effect. For example: `T: ~const Trait` (off) implies `T: Trait` (on)
```

Comment:
effects and subtyping seems very sus. What's the default variance for effect parameters?